File: problems_on_please/CpyAug2012/contests/day06/comb1/solutions/solution.py
#!/usr/bin/python3
#!/usr/bin/python3

fin = open('comb1.in', 'r')
fout = open('comb1.out', 'w')

def gen(n, k, prefix):
    if n == 0:
        print(" ".join(map(str, prefix)), file=fout)
        return
    if k < n:
        gen(n - 1, k, prefix + [0])
    if k > 0:
        gen(n - 1, k - 1, prefix + [1])


n, k = map(int, fin.read().strip().split())
gen(n, k, [])

File: problems_on_please/CpyAug2012/contests/day06/comb1/solutions/solution.cpp
#include<iostream>
#include <cstdio>

using namespace std;

void Gen(int n, int k, string Prefix)
{
	if(n==0)
		cout<<Prefix<<endl;
	else
	{
		if(n>k)
			Gen(n-1, k,Prefix+"0 ");
		if(k>0)
			Gen(n-1, k-1,Prefix+"1 ");
	}
}

int main()
{
    freopen("comb1.in", "r", stdin);
    freopen("comb1.out", "w", stdout);
	int n, k;
	cin>>n>>k;
	Gen(n,k,"");
	return 0;
}
